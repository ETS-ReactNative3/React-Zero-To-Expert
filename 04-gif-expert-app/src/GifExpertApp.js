import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

const GifExpertApp = ({ defaultCategories = [] }) => {
  // const categories = ['One Punch', ' Samurai X', 'Dragon Ball']
  // const [categories, setCategories] = useState(['Dragon Ball'])
  const [categories, setCategories] = useState(defaultCategories)

  // const handleAdd = () => {
  //   // Estas 2 soluciones hacen lo mismo
  //   // setCategories([...categories, 'HunterXHunter'])

  //   setCategories((cats) => [...categories, 'HunterXHunter'])
  // }

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />

      <hr />
      <br />
      {categories}
      <br />
      {/* <button onClick={handleAdd}>Agregar</button> */}
      <ol>
        {categories.map((category) => (
          <GifGrid key={category} category={category} />
        ))}
        {/* {categories.map((category) => (
          <li key={category}>{category}</li>
        ))} */}
      </ol>
    </>
  )
}

export default GifExpertApp
