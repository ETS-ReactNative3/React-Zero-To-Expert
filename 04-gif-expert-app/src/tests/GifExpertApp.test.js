import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import GifExpertApp from '../GifExpertApp'

describe('Pruebas en <GifExpertApp />', () => {
  const categories = ['Dragon Ball', 'One Punch']

  test('debe mostrarse correctamente', () => {
    const wrapper = shallow(<GifExpertApp />)
    expect(wrapper).toMatchSnapshot()
  })
  test('debe de mostrar una lista de categorias', () => {
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)

    expect(wrapper).toMatchSnapshot()
    // this will send error
    // expect(wrapper.find('GifGrid').length).toBe(categories.length + 1)
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
