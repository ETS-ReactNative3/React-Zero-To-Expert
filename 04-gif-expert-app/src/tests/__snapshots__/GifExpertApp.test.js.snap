// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <br />
  Dragon Ball
  One Punch
  <br />
  <ol>
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <br />
  <br />
  <ol />
</Fragment>
`;
